test/e2e/samples_test.go

```diff
@@ -116,7 +116,9 @@ func createBuilders(t *testing.T, decoder *helper.YAMLDecoder, sampleFile, testN
 				WithLogsMonitoring(logRefs...).
 				WithRestrictedSecurityContext().
 				WithLabel(run.TestNameLabel, fullTestName).
-				WithPodLabel(run.TestNameLabel, fullTestName)
+				WithPodLabel(run.TestNameLabel, fullTestName).
+				WithTestStorageClass()
```

Comment:
We already have the same kind of exception for Elasticsearch on EKS with [#6588](https://github.com/elastic/cloud-on-k8s/pull/6588/files) to tests samples due to #6515.

Shouldn't we make it conditional to only do it on `eks`?

Reply:
So I think the problem is for the Elasticsearch builder we add the `e2e-default` storage class in the `WithNodeSet` builder method which is used in all but the samples and recipes tests IIUC. We don't have an equivalent method for Logstash. That's why I added `WithTestStorageClass` into the constructor and here because the samples sidestep the builder methods and the constructor call is too early to have the right effect. 

What we could do is parameterise `WithTestStorageClass` on the platform so that it only adds the storage class for EKS runs. That would mean all other runs always use network attached storage.

The other option would be to just always run the Logstash tests with `e2e-default` which is what I am proposing here. I think it might be a bit faster overall and requires less conditional logic.

Reply:
I have no issues defaulting to the default storage class for logstash here. It can still be overridden conditionally per provider, so it isn't limiting us.

Reply:
> What we could do is parameterise WithTestStorageClass on the platform so that it only adds the storage class for EKS runs.
That's what I had in mind, but I'm ok with the current approach.